core/src/main/scala/akka/persistence/cassandra/journal/CassandraRecovery.scala

```diff
@@ -28,8 +28,12 @@ trait CassandraRecovery extends CassandraTagRecovery with TaggedPreparedStatemen
     new CassandraJournal.EventDeserializer(context.system)
 
   private[akka] def asyncReadHighestSequenceNrInternal(persistenceId: String, fromSequenceNr: Long): Future[Long] = {
-    asyncHighestDeletedSequenceNumber(persistenceId).flatMap { h =>
-      asyncFindHighestSequenceNr(persistenceId, math.max(fromSequenceNr, h), targetPartitionSize)
+    if (metadataEnabled) {
```

Comment:
This is already supported via `support-delete` https://github.com/akka/akka-persistence-cassandra/blob/master/core/src/main/resources/reference.conf#L111

Reply:
Oh, have not notice this feature in last changelogs, really missed it

> Need to remove the metadata change but otherwise LGTM.
> 
> What are causing these gaps? Unless you have events from an old version of the plugin there should
> not be any gaps for events by persistenceId

Regarding gaps - at least our production environment is far from ideal, any service always has a chance to corrupt pipelines on Broker\Processing level and affect our cluster
In last 8 months we had 3 cases like that, and only viable and fast solution is to cut out the bad part of event log and create new snapshots - anything else will cause major downtimes